micrometer-tracing/src/main/java/io/micrometer/tracing/handler/PropagatingSenderTracingObservationHandler.java

```diff
@@ -100,7 +100,11 @@ public void onStop(T context) {
         Span span = getRequiredSpan(context);
         tagSpan(context, span);
         customizeSenderSpan(context, span);
-        span.name(context.getContextualName() != null ? context.getContextualName() : context.getName());
+        // TODO do something else when getContextualName and getName are null?
+        String name = context.getContextualName() != null ? context.getContextualName() : context.getName();
+        if (name != null) {
+            span.name(name);
+        }
```

Comment:
While this is different logic than before, I think it is better and it respects the defined nullability. Related to https://github.com/micrometer-metrics/micrometer/issues/6417
Same for PropagatingReceiverTracingObservationHandler.